### PR TITLE
Refactor node layout to handle arbitrary widget sizes, add `Table.new` widget stub

### LINKED ISSUE #10484: Resizable widget for Table.new
First step of implementing [#10293](https://github.com/enso-org/enso/issues/10293)

Create widget which matches `Table.new` with simple fields (no cells like `2 + 2`) being only a top-level node's call. The widget should not display anything in particular in this task, but it should be resizable (and node should change size as well).

### PR DESCRIPTION
**Pull request by Frizi**
### Pull Request Description

Implements #10484, also fixed an issue with dropdown arrow icon being pointed in the wrong direction.

Changed the way we handle spacing around ports and other rounded widgets. Now it is the innermost token element that actually pads itself when appropriate, allowing rounded widgets to stay tightly nested together. This cleans up an issue we've had with an unnecessary padding at the end of node, and makes margins easier to control in general.

image

### Checklist

Please ensure that the following checklist has been satisfied before submitting the PR:

- [x] The documentation has been updated, if necessary.
- [x] Screenshots/screencasts have been attached, if there are any visual changes. For interactive or animated visual changes, a screencast is preferred.
- [x] All code follows the
      [Scala](https://github.com/enso-org/enso/blob/develop/docs/style-guide/scala.md),
      [Java](https://github.com/enso-org/enso/blob/develop/docs/style-guide/java.md),
      [TypeScript](https://github.com/enso-org/enso/blob/develop/docs/style-guide/typescript.md),
      and
      [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md)
      style guides. In case you are using a language not listed above, follow the [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md) style guide.
- [x] Unit tests have been written where possible.

### CODE REVIEW

File: app/gui2/src/components/GraphEditor/GraphNode.vue [hunk 1]
```diff
@@ -361,6 +345,7 @@ function startEditingNode(position?: Vec2 | undefined) {
 function getRelatedSpanOffset(domNode: globalThis.Node, domOffset: number): number {
   if (domNode instanceof HTMLElement && domOffset == 1) {
     const offsetData = domNode.dataset.spanStart
+    console.log('domNode', domNode)
```
Comment (kazcw) [inline app/gui2/src/components/GraphEditor/GraphNode.vue]: Logging

File: app/gui2/src/components/GraphEditor/NodeWidget.vue [hunk 2]
```diff
@@ -118,4 +119,31 @@ const spanStart = computed(() => {
...
+.NodeWidget.MarginBoundary {
+  > * {
+    --hard-edge-on-right: 0;
+  }
+}
+
+/* .MarginBoundary,
+.NodeWidget:not(.NodeWidget:has(.NodeWidget)) {
+  margin-right: calc(8px * var(--hard-edge-on-right) * var(--widget-is-leaf));
+} */
```
Comment (kazcw) [inline app/gui2/src/components/GraphEditor/NodeWidget.vue]: Commented

File: app/gui2/src/components/GraphEditor/widgets/WidgetFunction.vue [hunk 3]
```diff
@@ -167,17 +162,12 @@ function handleArgUpdate(update: WidgetUpdate): boolean {
 
         // Traverse the application chain, starting from the outermost application and going
         // towards the innermost target.
-        for (let innerApp of applications) {
+        for (let innerApp of [...app.iterApplications()]) {
```
Comment (kazcw) [inline app/gui2/src/components/GraphEditor/widgets/WidgetFunction.vue]:
```suggestion
        for (let innerApp of app.iterApplications()) {
```

File: app/gui2/src/components/GraphEditor/widgets/WidgetFunction.vue [hunk 4]
```diff
@@ -167,17 +162,12 @@ function handleArgUpdate(update: WidgetUpdate): boolean {
...
             // Found the application with the argument to remove. Skip the argument and use the
             // application target's code. This is the final iteration of the loop.
             const appTree = edit.getVersion(argApp.appTree)
-            if (graph.db.isNodeId(appTree.externalId)) {
-              // If the modified application is a node root, preserve its identity and metadata.
-              appTree.replaceValue(appTree.function.take())
-            } else {
-              appTree.replace(appTree.function.take())
-            }
+            appTree.replace(appTree.function.take())
```
Comment (kazcw) [inline app/gui2/src/components/GraphEditor/widgets/WidgetFunction.vue]: Doesn't changing the node ID cause problems with selection etc?

**Review by kazcw — APPROVED** [review]

File: app/gui2/src/components/GraphEditor/widgets/WidgetFunction.vue (hunk 3 above)
Comment (Frizi) [reply app/gui2/src/components/GraphEditor/widgets/WidgetFunction.vue]: That whole block of changes was accidental. A rebase caused a revert of changes from my previous branch and I missed that. Corrected now.

**Comment by Frizi** [comment]
Brought back proper handling for icons next to static dispatch access chains
image

**Pull request merged**